includes/class-coblocks-form.php

```diff
@@ -588,10 +590,37 @@ public function render_field_checkbox( $atts ) {
 
 		$label      = isset( $atts['label'] ) ? $atts['label'] : __( 'Select', 'coblocks' );
 		$label_slug = $checkbox_count > 1 ? sanitize_title( $label . '-' . $checkbox_count ) : sanitize_title( $label );
+		$required   = ( isset( $atts['required'] ) && $atts['required'] );
+
+		if ( $checkbox_count === 1 && $required ) {
+
+			wp_enqueue_script(
+				'coblocks-checkbox-required',
+				CoBlocks()->asset_source( 'js' ) . 'coblocks-checkbox-required.js',
+				array( 'jquery' ),
+				COBLOCKS_VERSION,
+				true
+			);
+
+		}
 
 		ob_start();
 
-		print( '<div class="coblocks-field">' );
+		printf(
+			'<div class="coblocks-field checkbox%1$s">
+				%2$s',
+			$required ? esc_attr( ' required' ) : '',
+			$required ? sprintf(
+				'<div class="required-error hidden">%s</div>',
+				/**
+				 * Filter the checkbox required text that displays when no checkbox is
+				 * selected when the form is submitted.
+				 *
+				 * @param string $error_text Error text displayed to the user.
+				 */
+				(string) apply_filters( 'coblocks_form_checkbox_required_text', esc_html__( 'Please select at least one checkbox.', 'coblocks' ) )
```

Comment:
If we are adding a new filter then we should also update the Docs to include information on this filter and how to apply it.